File: sem-01/informatics/06/python/08.py
def is_prime(num):
    if num < 2:
        return False

    for i in range(2, num // 2 + 1):
        if num % i == 0:
            return False

    return True

number = int(input("Введите число для проверки на простоту: "))

if is_prime(number):
    print(f"{number} - простое число.")
else:
    print(f"{number} - не простое число.")

File: sem-01/informatics/06/c++/08.cpp
#include <iostream>

bool isPrime(int num) {
    if (num < 2) {
        return false;
    }

    for (int i = 2; i <= num / 2; i++) {
        if (num % i == 0) {
            return false;
        }
    }

    return true;
}

int main() {
    int number;

    std::cout << "Введите число для проверки на простоту: ";
    std::cin >> number;

    if (isPrime(number)) {
        std::cout << number << " - простое число." << std::endl;
    } else {
        std::cout << number << " - не простое число." << std::endl;
    }

    return 0;
}
